app/models/user.rb

```diff
@@ -20,45 +18,17 @@ class User < ApplicationRecord
       attr_accessor :_skip_creating_profile
 
       # All new users should automatically have a profile
-      after_create_commit :create_profile, unless: :_skip_creating_profile
+      after_create_commit -> { Profile.create(user: self, data: Profiles::ExtractData.call(self)) },
+                          unless: :_skip_creating_profile
 
       # Keep saving changes locally for the time being, but propagate them to profiles.
       after_update_commit do
-        if (previous_changes.keys.map(&:to_sym) & User::PROFIE_FIELDS).present?
+        if previous_changes.keys.any? { |attribute| attribute.in?(Profile.mapped_attributes) }
```

Comment:
I'm not thrilled about this, but it's better than duplicating all this information in a constant here as it was previously.